source/core/ScriptDomain.cs

```diff
@@ -0,0 +1,908 @@
+		private List<IntPtr> _pinnedStrings = new List<IntPtr>();
+		private List<Tuple<string, Type>> _scriptTypes = new List<Tuple<string, Type>>();
+		private bool _recordKeyboardEvents = true;
+		private bool[] _keyboardState = new bool[256];
+		private bool disposed = false;
+		#endregion
+
+		public ScriptDomain()
+		{
+			this._appdomain = System.AppDomain.CurrentDomain;
+			this._executingThreadId = Thread.CurrentThread.ManagedThreadId;
+
+			sCurrentDomain = this;
+
+			this._appdomain.AssemblyResolve += new ResolveEventHandler(HandleResolve);
+			this._appdomain.UnhandledException += new UnhandledExceptionEventHandler(HandleUnhandledException);
+
+			Log("[INFO]", "Created new script domain with v", typeof(ScriptDomain).Assembly.GetName().Version.ToString(3), ".");
+
+			this._console = new ConsoleScript();
+		}
+		public void Dispose()
+		{
+			Dispose(true);
+			GC.SuppressFinalize(this);
+		}
+		protected virtual void Dispose(bool disposing)
+		{
+			if (!disposed)
+			{
+				if (disposing)
+				{
+				}
+				CleanupStrings();
+				disposed = true;
+			}
+		}
+		~ScriptDomain()
+		{
+			Dispose(false);
+		}
+
+		public static Script ExecutingScript
+		{
+			get
+			{
+				if (ReferenceEquals(sCurrentDomain, null))
+				{
+					return null;
+				}
+
+				return sCurrentDomain._executingScript;
+			}
+		}
+		public static ScriptDomain CurrentDomain
+		{
+			get
+			{
+				return sCurrentDomain;
+			}
+		}
+		public string Name
+		{
+			get
+			{
+				return _appdomain.FriendlyName;
+			}
+		}
+		public AppDomain AppDomain
+		{
+			get
+			{
+				return _appdomain;
+			}
+		}
+		public ConsoleScript Console
+		{
+			get
+			{
+				return _console;
+			}
+		}
+		public Script[] RunningScripts
+		{
+			get
+			{
+				return _runningScripts.ToArray();
+			}
+		}
+
+		public static ScriptDomain Load(string path)
+		{
+			if (!System.IO.Path.IsPathRooted(path))
+			{
+				path = System.IO.Path.Combine(System.IO.Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), path);
+			}
+
+			path = System.IO.Path.GetFullPath(path);
+
+			// Clear log
+			string logPath = System.IO.Path.ChangeExtension(Assembly.GetExecutingAssembly().Location, ".log");
+
+			try
+			{
+				System.IO.File.WriteAllText(logPath, string.Empty);
+			}
+			catch (Exception ignored)
+			{ }
+
+			// Create AppDomain
+			var setup = new AppDomainSetup();
+			setup.ApplicationBase = path;
+			setup.ShadowCopyFiles = "true";
+			setup.ShadowCopyDirectories = path;
+
+			var appdomain = System.AppDomain.CreateDomain("ScriptDomain_" + (path.GetHashCode() * Environment.TickCount).ToString("X"), null, setup, new System.Security.PermissionSet(System.Security.Permissions.PermissionState.Unrestricted));
+			appdomain.InitializeLifetimeService();
+
+			ScriptDomain scriptdomain = null;
+
+			try
+			{
+				scriptdomain = (ScriptDomain)(appdomain.CreateInstanceFromAndUnwrap(typeof(ScriptDomain).Assembly.Location, typeof(ScriptDomain).FullName));
```

Comment:
There is something wrong with this line. Any solutions? I need to solve this problem before testing any further .